### ISSUE TITLE
it's possible to add a scheduled monitoring feature to activate file monitoring at a specific time. 

### ISSUE THREAD

**Issue by ilanni2460** [michael-doubez/filestat_exporter#32]
it's possible to add a scheduled monitoring feature to activate file monitoring at a specific time.

**Comment by michael-doubez** [comment]
No. Prometheus and service discovery are the compenents that are timing the monitoring

**Comment by ilanni2460** [comment]
@michael-doubez Thank you very much for your reply, but I found out whether another exporter that monitors sql statements supports timing to grab data.

This is url :
https://github.com/albertodonato/query-exporter

`
schedule:
a schedule for executing queries at specific times.
This is expressed as a Cron-like format string (e.g. */5 * * * * to run every five minutes).
If a value is specified for schedule, an interval can't be specified.
If no value is specified (or specified as null), the query is only executed upon HTTP requests.
`

I looked at it, and her implementation principle is to capture metrics at a specified time, and the metrics cannot be captured at a non-specified time.

**Comment by michael-doubez** [comment]
Actually, it makes sense in case of a slow NFS by example.

**Comment by michael-doubez** [comment]
I've added the tree feature in #42 .
I am planinng on adding a url parameter to metrics collection such that different tree (root) can be polled at different speed.

That way, one can poll /slow/ filesystems with an interval of ~4minutes (with a bigger timeout) while polling other more often.
This is not really what you asked for but keeping cache of metrics is a snake pit.

It you really need to take snapshots, I could add a feature for exporting metrics to a pushgateway.

**Comment by ilanni2460** [comment]
@michael-doubez Very good, I have been paying attention to this feature

**Comment by michael-doubez** [comment]
Sorry for the delay.

Prometheus client is not friendly with url parameters - it will take time

**Comment by michael-doubez** [comment]
Too old

**Comment by ilanni2460** [comment]
@michael-doubez it is ok. Take you time.
